src/server/engine_shard_set.cc

```diff
@@ -39,8 +52,45 @@ using absl::GetFlag;
 
 namespace {
 
+constexpr DbIndex DEFAULT_DB_INDEX = 0;
+
 vector<EngineShardSet::CachedStats> cached_stats;  // initialized in EngineShardSet::Init
 
+inline bool MemUsageChanged(uint64_t current, uint64_t now) {
+  static const float commit_use_threshold = GetFlag(FLAGS_commit_use_threshold);
+  // we only care whether the memory usage is going up
+  // otherwise we don't really case
+  return (current * commit_use_threshold) < float(now);
+}
+
+// This function checks 3 things:
```

Comment:
see the comments about what this does, but mainly we want to ensure that we are not running when there is no reason to run